Examples/Algorithms/Geant4HepMC/src/EventRecording.cpp

```diff
@@ -73,14 +74,66 @@ ActsExamples::ProcessCode ActsExamples::EventRecording::execute(
     // Begin with the simulation
     m_runManager->BeamOn(1);
 
+    // Test if the event was aborted
+    if (SteppingAction::instance()->eventAborted()) {
+      continue;
+    }
+
+    // Set event start time
     HepMC3::GenEvent event = ActsExamples::EventAction::instance()->event();
-    HepMC3::FourVector shift(0., 0., 0., part.time());
+    HepMC3::FourVector shift(0., 0., 0., part.time() / Acts::UnitConstants::s);
     event.shift_position_by(shift);
 
-    // Store the result
-    events.push_back(std::move(event));
+    // Set beam particle properties
+    HepMC3::FourVector beamMom4(part.momentum4()[0], part.momentum4()[1],
+                                part.momentum4()[2], part.momentum4()[3]);
+    auto beamParticle = event.particles()[0];
+    beamParticle->set_momentum(beamMom4);
+    beamParticle->set_pid(part.pdg());
+
+    if (m_cfg.eventSelectionProcess.empty()) {
+      // Store the result
+      events.push_back(std::move(event));
+    } else {
+      bool storeEvent = false;
+      // Test if the event has a process of interest in it
+      for (const auto& vertex : event.vertices()) {
+        if (vertex->id() == -1) {
+          vertex->add_particle_in(beamParticle);
+        }
+        const std::vector<std::string> vertexAttributes =
```

Comment:
The vector is built inside `attribute_names`. I cannot make a reference to this rvalue.

Reply:
Just as a comment: you can create a const reference to a temporary value, e.g. the return value of a function. This will prolong the lifetime of the referenced value until the const reference goes out-of-scope (same as it would if you store the value). The nice thing is that this allows you to not care whether the function returns by value or by const-reference. You can also receive the return value in a const reference and ensure that there are no unnecessary copies (even if somebody decides to change the return type in the future).